src/User.ts

```diff
@@ -178,6 +182,194 @@ export class User extends PrizePool {
     }
   }
 
+  /**
+   * Requests a signature from the user to approve a deposit
+   * @param amountUnformatted an unformatted and decimal shifted amount to approve for deposits
+   * @param customDeadline a custom deadline to override the default (5 mins from signature)
+   * @returns a promise to request a signature
+   */
+  async getPermitAndDepositSignaturePromise(
+    amountUnformatted: BigNumber,
+    customDeadline?: number
+  ): Promise<(ERC2612PermitMessage & RSV) | undefined> {
+    const errorPrefix = 'User [approveDepositsWithSignature]'
+    await this.validateSignerNetwork(errorPrefix)
+
+    const tokenContract = await this.getTokenContract({ eip2612: true })
+    const tokenData = await this.getTokenData()
+
+    if (
+      !this.eip2612PermitAndDepositMetadata ||
+      !this.eip2612PermitAndDepositContract ||
+      !this.tokenMetadata ||
+      !this.signer.provider
+    )
+      throw new Error(errorPrefix + ` | Error initializing contract metadata.`)
+
+    const usersAddress = await this.signer.getAddress()
+
+    const domain = {
+      name: tokenData.name,
+      version: this.tokenMetadata.version.major.toString(),
+      chainId: this.chainId,
+      verifyingContract: this.tokenMetadata.address
+    }
+
+    // NOTE: Nonce must be passed manually for signERC2612Permit to work with WalletConnect
+    const deadline =
+      customDeadline ?? (await this.signer.provider.getBlock('latest')).timestamp + 5 * 60
+    const response = await tokenContract.functions.nonces(usersAddress)
+    const nonce: BigNumber = response[0]
+
+    const signaturePromise = signERC2612Permit(
+      this.signer,
+      domain,
+      usersAddress,
+      this.eip2612PermitAndDepositMetadata.address,
+      amountUnformatted.toString(),
+      deadline,
+      nonce.toNumber()
+    )
+
+    return signaturePromise
+  }
+
+  /**
+   * Requests a signature from the user to approve a delegation
+   * @param delegateTo an address to delegate to, if different from the users' wallet
+   * @param customDeadline a custom deadline to override the default (5 mins from signature)
+   * @returns a promise to request a signature
+   */
+  async getPermitAndDelegateSignaturePromise(
+    delegateTo?: string,
+    customDeadline?: number
+  ): Promise<(ERC2612TicketPermitMessage & RSV) | undefined> {
+    const errorPrefix = 'User [approveDelegationWithSignature]'
+    await this.validateSignerNetwork(errorPrefix)
+
+    const ticketContract = await this.getTicketContract()
+
+    if (
+      !this.eip2612PermitAndDepositMetadata ||
+      !this.eip2612PermitAndDepositContract ||
+      !this.ticketMetadata ||
+      !this.signer.provider
+    )
+      throw new Error(errorPrefix + ` | Error initializing contract metadata.`)
+
+    const usersAddress = await this.signer.getAddress()
+
+    const domain = {
```

Comment:
I'd add a note here about the hardcoded values in our contracts just for future reference

Reply:
Added in d8f3a93